clients/js/test/setMint.test.ts

```diff
@@ -0,0 +1,205 @@
+    umi,
+    inscriptionMetadataAccount
+  );
+
+  t.like(inscriptionMetadata, <InscriptionMetadata>{
+    key: Key.MintInscriptionMetadataAccount,
+    inscriptionAccount: inscriptionAccount[0],
+    bump: inscriptionMetadataAccount[1],
+    dataType: DataType.Uninitialized,
+    updateAuthorities: [umi.identity.publicKey],
+    associatedInscriptions: [] as AssociatedInscription[],
+    mint: some(mint.publicKey),
+  });
+});
+
+test('it cannot set the mint on an Inscription account', async (t) => {
+  // Given a Umi instance and a new signer.
+  const umi = await createUmi();
+  umi.use(mplTokenMetadata());
+
+  const mint = generateSigner(umi);
+  await createV1(umi, {
+    mint,
+    name: 'My NFT',
+    uri: 'https://arweave.net/LcjCf-NDr5bhCJ0YMKGlc8m8qT_J6TDWtIuW8lbu0-A',
+    sellerFeeBasisPoints: percentAmount(5.5),
+    tokenStandard: TokenStandard.NonFungible,
+  }).sendAndConfirm(umi);
+
+  await mintV1(umi, {
+    mint: mint.publicKey,
+    tokenStandard: TokenStandard.NonFungible,
+  }).sendAndConfirm(umi);
+
+  const inscriptionAccount = generateSigner(umi);
+
+  const inscriptionMetadataAccount = await findInscriptionMetadataPda(umi, {
+    inscriptionAccount: inscriptionAccount.publicKey,
+  });
+
+  let builder = new TransactionBuilder();
+
+  // When we create a new account.
+  builder = builder.append(
+    initialize(umi, {
+      inscriptionAccount,
+    })
+  );
+
+  // Set the mint on the account.
+  builder = builder.append(
+    setMint(umi, {
+      mintInscriptionAccount: inscriptionAccount.publicKey,
+      inscriptionMetadataAccount,
+      mintAccount: mint.publicKey,
+    })
+  );
+
+  const promise = builder.sendAndConfirm(umi);
+  // Then an error is thrown.
+  await t.throwsAsync(promise, { name: 'DerivedKeyInvalid' });
+});
+
+test('it cannot set the wrong mint on a Mint Inscription account', async (t) => {
+  // Given a Umi instance and a new signer.
+  const umi = await createUmi();
+  umi.use(mplTokenMetadata());
+
+  const mint = generateSigner(umi);
+  await createV1(umi, {
+    mint,
+    name: 'My NFT',
+    uri: 'https://arweave.net/LcjCf-NDr5bhCJ0YMKGlc8m8qT_J6TDWtIuW8lbu0-A',
+    sellerFeeBasisPoints: percentAmount(5.5),
+    tokenStandard: TokenStandard.NonFungible,
+  }).sendAndConfirm(umi);
+
+  await mintV1(umi, {
+    mint: mint.publicKey,
+    tokenStandard: TokenStandard.NonFungible,
+  }).sendAndConfirm(umi);
+
+  const wrongMint = generateSigner(umi);
+  await createV1(umi, {
+    mint: wrongMint,
+    name: 'My NFT',
+    uri: 'https://arweave.net/LcjCf-NDr5bhCJ0YMKGlc8m8qT_J6TDWtIuW8lbu0-A',
+    sellerFeeBasisPoints: percentAmount(5.5),
+    tokenStandard: TokenStandard.NonFungible,
+  }).sendAndConfirm(umi);
+
+  await mintV1(umi, {
+    mint: wrongMint.publicKey,
+    tokenStandard: TokenStandard.NonFungible,
+  }).sendAndConfirm(umi);
+
+  const inscriptionAccount = await findMintInscriptionPda(umi, {
+    mint: mint.publicKey,
+  });
```

Comment:
Also here I think there should be another "And..." comment saying something about how there's a different mint PDA being sent.